factory/random.py

```diff
@@ -10,7 +10,11 @@
 
 def get_random_state():
     """Retrieve the state of factory.fuzzy's random generator."""
-    return randgen.getstate()
+    state = randgen.getstate()
+    randgen.state_set = True
```

Comment:
Ok, now I see what's going on. Makes sense, but I am a bit concerned about `get_random_state` producing side-effects. What's is this function used for? It would help me understand some of the use cases.

Reply:
I copied it from `set_random_state()`.
It triggers a warning in `FuzzyDate` and `FuzzyDateTime` subclasses.

https://github.com/FactoryBoy/factory_boy/blob/541f6e8c6e7ea16fbff7907338e82afc17de3d95/factory/fuzzy.py#L18-L22

Since these factories are independent from Faker, there's no need to issue that warning when `get_random_state()` was called. I'll remove it, thanks for pointing it out.

Reply:
AFAICT, it's used to control random numbers generation. One can first acquire the current state of the random number generator and reset it.

https://docs.python.org/3/library/random.html#random.getstate states:

> Return an object capturing the current internal state of the generator. This object can be passed to setstate() to restore the state.

I think that's the main use case.